### ISSUE TITLE
Apple Silicon - Routes-Controller Image Does Not Support ARM64

### ISSUE THREAD

**Issue by cgruver** [code-ready/crc#3365]
### General information

  * MacOS
  * vfkit

## CRC version
```bash
CRC version: 2.9.0+589ab2cd
OpenShift version: 4.11.3
```
  
The routes-controller image that runs in a cluster started with `crc` does not run on apple silicon.

The pod is in a crash loop state, and the executable in the pod appears to be compiled for amd64, not arm64.

I'm going to fork the code for routes-controller and verify.

**Comment by praveenkumar** [comment]
We had those images to support arm64 https://quay.io/repository/crcont/routes-controller?tab=tags but looks like latest one didn't build for multiarch, we will take a look.

**Comment by cgruver** [comment]
I was able to get the `routes-controller` pod running, but it requires a work-around in the `4.11.3` bundle.  The work-around is necessary because the pod has an `imagePullPolicy` of `IfNotPresent` and the incorrect container image is already in the bundle.

1. Start OpenShift Local

```bash
crc setup
crc start
```

1. After the cluster has started, log in with `kubeadmin` credentials.

1. Observe that `routes-controller` pod in the `openshift-ingress` namespace is in a crash loop.

1. Retrieve the pod spec in YAML:

```bash
oc get pod routes-controller -n openshift-ingress -o yaml > /tmp/routes-controller.yaml
```

1. Edit `/tmp/routes-controller.yaml` removing the boilerplate added by OpenShift and changing the `imagePullPolicy` from `IfNotPresent` to `Always`

1. Delete the pod:

```bash
oc delete pod routes-controller -n openshift-ingress
```

1. Recreate it with the file:

```bash
oc apply -n openshift-ingress -f /tmp/routes-controller.yaml
```

1. Observe that the `routes-controller` pod is running.

Here's an example:

```yaml
kind: Pod
apiVersion: v1
metadata:
  name: routes-controller
  namespace: openshift-ingress
  annotations:
    k8s.v1.cni.cncf.io/network-status: |-
      [{
          "name": "openshift-sdn",
          "interface": "eth0",
          "ips": [
              "10.217.0.49"
          ],
          "default": true,
          "dns": {}
      }]
    k8s.v1.cni.cncf.io/networks-status: |-
      [{
          "name": "openshift-sdn",
          "interface": "eth0",
          "ips": [
              "10.217.0.49"
          ],
          "default": true,
          "dns": {}
      }]
    openshift.io/scc: anyuid
spec:
  restartPolicy: Always
  serviceAccountName: router
  imagePullSecrets:
    - name: router-dockercfg-m5kjz
  priority: 0
  schedulerName: default-scheduler
  enableServiceLinks: true
  terminationGracePeriodSeconds: 30
  preemptionPolicy: PreemptLowerPriority
  nodeName: crc-j2d48-master-0
  securityContext:
    seLinuxOptions:
      level: 's0:c24,c19'
  containers:
    - name: routes-controller
      image: 'quay.io/crcont/routes-controller:latest'
      resources: {}
      volumeMounts:
        - name: kube-api-access-4jgff
          readOnly: true
          mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      terminationMessagePath: /dev/termination-log
      terminationMessagePolicy: File
      imagePullPolicy: Always
      securityContext:
        capabilities:
          drop:
            - MKNOD
  serviceAccount: router
  volumes:
    - name: kube-api-access-4jgff
      projected:
        sources:
          - serviceAccountToken:
              expirationSeconds: 3607
              path: token
          - configMap:
              name: kube-root-ca.crt
              items:
                - key: ca.crt
                  path: ca.crt
          - downwardAPI:
              items:
                - path: namespace
                  fieldRef:
                    apiVersion: v1
                    fieldPath: metadata.namespace
          - configMap:
              name: openshift-service-ca.crt
              items:
                - key: service-ca.crt
                  path: service-ca.crt
        defaultMode: 420
  dnsPolicy: ClusterFirst
  tolerations:
    - key: node.kubernetes.io/not-ready
      operator: Exists
      effect: NoExecute
      tolerationSeconds: 300
    - key: node.kubernetes.io/unreachable
      operator: Exists
      effect: NoExecute
      tolerationSeconds: 300
EOF
```

**Comment by praveenkumar** [comment]
@cgruver Thanks, in next release the bundle we will release should not have this issue.

**Comment by cgruver** [comment]
Fixed in `2.10.1`
